app/models/event.rb

```diff
@@ -17,4 +18,22 @@ def before_status_full
   def after_status_full
     STATUSES[after_status]
   end
+
+  def new_for?(user)
+    notifications.where('created_at > ? and user_id = ?',
```

Comment:
why not 
user.notifications.where('created_at > ?',  user.last_check_notifications_at).any?